records_mover/db/mysql/load_options.py

```diff
@@ -0,0 +1,289 @@
+    # through due to our use of "type: ignore" rather than run-time
+    # validation below.
+    def invalid_hint(fail_if_cant_handle_hint: bool,
+                     hint_name: str,
+                     hints: RecordsHints,
+                     the_bad_hint: NoReturn) -> NoReturn:
+        cant_handle_hint(fail_if_cant_handle_hint, hint_name, hints)
+
+
+MYSQL_CHARACTER_SETS_FOR_LOAD: Dict[HintEncoding, MySqlCharacterSet] = {
+    "UTF8": 'utf8',
+    "UTF16": 'utf16',
+    "UTF16LE": 'utf16le',
+    # https://dev.mysql.com/doc/refman/8.0/en/charset-unicode.html
+    #
+    # The MySQL implementation of UCS-2, UTF-16, and UTF-32 stores
+    # characters in big-endian byte order and does not use a byte
+    # order mark (BOM) at the beginning of values. Other database
+    # systems might use little-endian byte order or a BOM. In such
+    # cases, conversion of values will need to be performed when
+    # transferring data between those systems and MySQL. The
+    # implementation of UTF-16LE is little-endian.
+    "UTF16BE": 'utf16',
+    # MySQL uses no BOM for UTF-8 values.
+    "LATIN1": 'latin1',
+    #  MySQL's latin1 is the same as the Windows cp1252 character set.
+    #
+    # https://dev.mysql.com/doc/refman/8.0/en/charset-we-sets.html
+    "CP1252": 'latin1',
+}
+
+
+# Mark as total=False so we can create this incrementally
+class MySqlLoadOptions(NamedTuple):
+    character_set: MySqlCharacterSet
+    fields_terminated_by: str  # default '\t'
+    fields_enclosed_by: Optional[str]  # default None
+    fields_optionally_enclosed_by: Optional[str]  # default None
+    fields_escaped_by: Optional[str]  # default '\'
+    lines_starting_by: str  # default ''
+    lines_terminated_by: str  # default '\n'
+    ignore_n_lines: int
+
+    def generate_load_data_sql(self,
+                               filename: str,
+                               schema_name: str,
+                               table_name: str) -> TextClause:
+        sql = f"""\
+LOAD DATA
+LOCAL INFILE :filename
+INTO TABLE {schema_name}.{table_name}
+CHARACTER SET :character_set
+FIELDS
+    TERMINATED BY :fields_terminated_by
+"""
+        if self.fields_enclosed_by is not None and self.fields_optionally_enclosed_by is None:
+            sql += "    ENCLOSED BY :fields_enclosed_by\n"
+        elif self.fields_enclosed_by is None and self.fields_optionally_enclosed_by is not None:
+            sql += "    OPTIONALLY ENCLOSED BY :fields_optionally_enclosed_by\n"
+        elif self.fields_enclosed_by is not None and self.fields_optionally_enclosed_by is not None:
+            raise SyntaxError('fields_enclosed_by and optionally_fields_enclosed_by '
+                              'cannot both be set')
+        if self.fields_escaped_by is not None:
+            sql += "    ESCAPED BY :fields_escaped_by\n"
+        sql += """\
+LINES
+    STARTING BY :lines_starting_by
+    TERMINATED BY :lines_terminated_by
+IGNORE :ignore_n_lines LINES
+"""
```

Comment:
Ruby has a nice heredoc syntax which lets you keep things indented without adding indentation to the actual string.  I couldn't find that for Python.